c-api/dotnet/src/Picky/Error.cs

```diff
@@ -0,0 +1,39 @@
+namespace Devolutions.Picky;
+
+public class ErrorException : System.Exception
+{
+    public ErrorException(String message) : base(message) { }
+}
+
+public static class Error
+{
+    /// <summary>Returns a String containing last error message from Picky.</summary>
+    public static String Last()
+    {
+        int len = Native.Raw.last_error_length_utf8();
+
+        if (len == 0)
+        {
+            throw new ErrorException("no error message to retrieve from Picky");
```

Comment:
Is this exceptional? It's a public function, I wouldn't `throw` in this case. Returning null has it's own warts but .NET developers love `StringIsNullOrEmpty` to deal with that. Otherwise, you could keep the exception but maybe expose a `HasError` or similar function too, that I can check before calling `Last`.

Reply:
That was actually the initial design, but I changed because I don't want the user to check "native" errors that way (a `HasError` function would have the same issue).
The reason is that if user forget, or deliberately _don't_ call the function after calling a native function that could have failed, the thread local static don't get cleared and another native call later that didn't failed could be considered as failed because of the error of the previous call.
As such, user is supposed to check if the native function returned a special value (null, -1…) and only then call this function to retrieve the error description. So I figured that maybe an exception would be appropriate.
What do you think? Maybe I should explain this in the function doc?

Reply:
> I don't expect user to use this often as we provide idiomatic wrappers around the native calls anyway

In that case, I would leave it as is!